File: app/src/main/java/com/leaguechampions/ui/championdetails/ChampionDetailsPresenterModule.java
package com.leaguechampions.ui.championdetails;

import dagger.Module;
import dagger.Provides;

@Module
public class ChampionDetailsPresenterModule {

    @Provides
    public ChampionDetailsPresenter.ChampionDetailsView provideView(ChampionDetailsActivity activity) {
        return activity;
    }
}

File: app/src/main/java/com/leaguechampions/ui/championdetails/ChampionDetailsPresenterModule.kt
package com.leaguechampions.ui.championdetails

import dagger.Module
import dagger.Provides

@Module
class ChampionDetailsPresenterModule {
    @Provides
    internal fun provideView(activity: ChampionDetailsActivity): ChampionDetailsPresenter.ChampionDetailsView {
        return activity
    }
}
